midterm-2023-11-06/code/5115-nikola-tosic/script.ts

```diff
@@ -0,0 +1,162 @@
+interface Album {
+    name: string;
+    year: number;
+}
+
+interface Band {
+    id: number;
+    name: string;
+    formed: number;
+    location: string;
+    country: string;
+    members: string[];
+    albums: Album[];
+}
+
+type BandSorter = (first: Band, second: Band) => number;
+
+document.addEventListener("DOMContentLoaded", siteCode);
+
+let bands: Band[] = [];
+
+async function siteCode() {
+    const data = await loadData();
+    bands = data;
+
+    fillCountries(bands);
+
+    displayBands(bands);
+
+    const nameSort = document.getElementById("sort-name")!;
+    nameSort.addEventListener("click", sortByName);
+
+    const idSort = document.getElementById("sort-id")!;
+    idSort.addEventListener("click", sortById);
+
+    const applyFilterButton = document.getElementById("apply-filter")!;
+    applyFilterButton.addEventListener("click", applyFilter);
+
+    const modal = document.getElementById("album-details")!;
+    modal.addEventListener("click", () => {
+        modal.classList.add("hidden");
+    });
+}
+
+const nameSorter: BandSorter = (first, second) => first.name.localeCompare(second.name);
+const idSorter: BandSorter = (first, second) => first.id - second.id;
+
+const sortByName = () => {
+    const sortedBands = bands.slice().sort(nameSorter);
+    displayBands(sortedBands);
+};
+
+const sortById = () => {
+    const sortedBands = bands.slice().sort(idSorter);
+    displayBands(sortedBands);
+};
+
+const fillCountries = (bands: Band[]) => {
```

Comment:
The structure of the method is ok, but the issue is that the `band` will **not** have a country property - so in essence we'll only get undefined. +5